OHIF/Viewers
# feat: Add customization support for UI components

### Context

- Created a custom hook and provider to manage and share services across the application.
- Added support for rendering custom UI components.
- This PR has been incorporated by [FlyWheel.io](https://flywheel.io/)



### Changes & Results

- Developed a new hook and provider for accessing services directly across the application, eliminating the need to pass them through props.
- By using the new CustomizableRenderingComponents, we made changes to the following components to support receiving custom components through the customization service.

1. LabellingFlow, 
2. LoadingIndicatorProgress, 
3. LoadingIndicatorTotalPercent, 
4. ProgressLoadingBar, 
5. ContextMenuItem,
6. ViewportActionCorners
 
- As part of this, we moved the default OSS components to the extensions/default and extensions/cornerstone folders and now send the default components through the customization module.
- This allows users to replace these components with their own customizations, either in mode or global customizations.



### Testing

- Set a new component through the customization service using the corresponding customizationId for the mentioned components.
- Observe the behavior in the viewer.
- The newly set component should replace the default OSS component.



### Checklist

#### PR



  semantic-release format and guidelines.

#### Code

  etc.)

#### Public Documentation Updates



  additions or removals.

#### Tested Environment

- [x] Node version: 18.19.0 
- [x] Browser: Chrome 131.0.6778.205
  


[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch
